src/empire/xml.c

```diff
@@ -312,6 +395,52 @@ static void xml_end_invasion_path(void)
     data.invasion_path_idx = 0;
 }
 
+static void xml_end_distant_battle_path(void)
+{
+    empire_object_type obj_type = 0;
+    int image_id = 0;
+    if (data.distant_battle_path_type == DISTANT_BATTLE_PATH_ROMAN) {
+        obj_type = EMPIRE_OBJECT_ROMAN_ARMY;
+        image_id = GROUP_EMPIRE_ROMAN_ARMY;
+    } else if (data.distant_battle_path_type == DISTANT_BATTLE_PATH_ENEMY) {
+        obj_type = EMPIRE_OBJECT_ENEMY_ARMY;
+        image_id = GROUP_EMPIRE_ENEMY_ARMY;
+    } else {
+        data.success = 0;
+        log_error("Invalid distant battle path type", 0, data.distant_battle_path_type);
+        return;
+    }
+
+    int month = 1;
+    for (int i = 1; i < data.distant_battle_wapoint_idx; i++) {
+        waypoint *last = &data.distant_battle_waypoints[i - 1];
+        waypoint *current = &data.distant_battle_waypoints[i];
+        int x_diff = current->x - last->x;
+        int y_diff = current->y - last->y;
+        for (int j = 0; j < current->num_months; j++) {
+            if (data.next_empire_obj_id >= MAX_EMPIRE_OBJECTS) {
```

Comment:
Won't this create a while lot of army objects, one per month? Is there really no simpler way to do this?

Reply:
This is how the rest of the code expects the empire to be defined. Trade routes have the same issue where every 3 dots are defined as a separate object. Since they’re purely visual, we were able to change them pretty easily. Honestly it’s not too bad though, worst case would be around 48 objects as you only have 2 years to reach the city in distress (12 months * 2 years * 2 armies (enemy and friendly)). Flipping through the existing empire setups in the scenario editor, the most I saw was 32.

Reply:
How inefficient! Okay, I guess since it's like that anyway we can keep it.

In the future though,  we should improve that.